### PR TITLE
Add CI checks as github actions

### PR DESCRIPTION
**Pull request by tpoliaw**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    - uses: Swatinem/rust-cache@v2
+    - name: Check formatting
+      run: cargo +nightly fmt -- --check
+    - name: Clippy
+      run: |
+        cargo --version
+        cargo clippy --version
+        cargo clippy --all-targets --all-features -- --deny warnings
+
+  sqlx_offline:
```
Comment (callumforrester) [inline .github/workflows/ci.yml:30]: Could: Give this check a more descriptive name, maybe `sql queries` (if I'm reading what it does right)?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  CARGO_TERM_COLOR: always
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: dtolnay/rust-toolchain@stable
+      with:
+          toolchain: nightly
```
[the remark this answers is not in the document]
Comment (tpoliaw) [reply .github/workflows/ci.yml:19]: cf. [the explanation from amygdala](https://alfred.diamond.ac.uk/documentation/amygdala/dev/reference.html#code-style)

File: .github/workflows/ci.yml (hunk 1 above)
Comment (tpoliaw) [reply .github/workflows/ci.yml:30]: It compares the query data used for building the project offline with the current schema - so kind of both? [Docs here](https://github.com/launchbadge/sqlx/tree/main/sqlx-cli#enable-building-in-offline-mode-with-query)

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
+name: Numtracker CI
```
[the remark this answers is not in the document]
Comment (tpoliaw) [reply .github/workflows/ci.yml:1]: I don't think either name really says what it does and numtracker was shorter. Agree there should probably only be one name though.

**Pull request merged**